plugins/database/mysql/mysql.go

```diff
@@ -0,0 +1,187 @@
+package mysql
+
+import (
+	"database/sql"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/vault/builtin/logical/database/dbplugin"
+	"github.com/hashicorp/vault/helper/strutil"
+	"github.com/hashicorp/vault/plugins/helper/database/connutil"
+	"github.com/hashicorp/vault/plugins/helper/database/credsutil"
+	"github.com/hashicorp/vault/plugins/helper/database/dbutil"
+)
+
+const defaultMysqlRevocationStmts = `
+	REVOKE ALL PRIVILEGES, GRANT OPTION FROM '{{name}}'@'%'; 
+	DROP USER '{{name}}'@'%'
+`
+const mySQLTypeName = "mysql"
+
+type MySQL struct {
+	connutil.ConnectionProducer
+	credsutil.CredentialsProducer
+}
+
+// New implements builtinplugins.BuiltinFactory
+func New() (interface{}, error) {
+	connProducer := &connutil.SQLConnectionProducer{}
+	connProducer.Type = mySQLTypeName
+
+	credsProducer := &credsutil.SQLCredentialsProducer{
+		DisplayNameLen: 4,
+		UsernameLen:    16,
```

Comment:
Could this be a plugin-specific configuration item? (I think we did end up supporting those?) 16 is _so_ small that it'd be really nice if we could default to 16 but allow larger (or maybe default to 32 and allow larger/smaller).